### PR TITLE
EDSC-797: Hide all filters but temporal for CWIC granules

### PR DESCRIPTION
**Pull request by mightynimble**

### CODE REVIEW

**Comment by mightynimble** [comment]
## Existing issues and questions of this pull request:

**ISSUES**

- The url doesn't have temporal param `pg[0][qt]`. So a refresh of the page will lose temporal constraint.
- CWIC endpoint doesn't accept encoded time parameters (a 400, see `FIXME` in `lib/echo/cwic_client.rb`.)

**QUESTIONS**
 - Should we support recurring temporal search? The CWIC endpoint doesn't support this feature.(http://cwic.wgiss.ceos.org/opensearch/datasets/gov.noaa.class.J2-XGDR/osdd.xml?clientId=edsc-dev)

**Comment by bilts** [comment]
If CWIC doesn't support recurring temporal search, we cannot support it. It should be disabled on the granule details view, and we need a new story to deal with what to do if the user applies collection-level recurring temporal.

File: app/assets/javascripts/models/data/collection.js.coffee [hunk 1]
```diff
@@ -259,7 +256,7 @@ ns.Collection = do (ko
       null
 
     canFocus: ->
-      @hasAtomData() && @granuleDatasourceName()
+      @hasAtomData() && @granuleDatasourceName()?
```
Comment (bilts) [inline app/assets/javascripts/models/data/collection.js.coffee]: Was there an issue that required the `?` to be added?

File: app/assets/javascripts/modules/banner.coffee [hunk 2]
```diff
@@ -27,7 +27,7 @@
       $banner.data('banner.persist', options.persist)
       $('body').after($banner)
       # Do this in a timeout so the element has time to be placed in the DOM and animations can happen
-      setTimeout((-> $banner.removeClass('banner-hidden')), 0)
+      setTimeout((-> $banner?.removeClass('banner-hidden')), 0)
```
Comment (bilts) [inline app/assets/javascripts/modules/banner.coffee]: See previous comment. The banner should be in the DOM. Is there a case where it's not?

File: app/assets/javascripts/modules/map/leaflet-plugins/interpolation.js.coffee [hunk 3]
```diff
@@ -22,13 +22,7 @@ ns.interpolation = do (L, gcInterpolate = window.edsc.map.geoutil.gcInterpolate,
...
     return [] if latLngs.length == 0
 
-    # Clone path and set its last element to its first so we can interpolate the last segment
-    if Math.abs(latLngs[0].lng) == Math.abs(latLngs[latLngs.length - 1].lng) == 180
-      # In this case the last element is an artificial longitude crossing.  Avoid interpolating
-      # with the first to prevent drawing strokes along the dateline
-      latLngs = latLngs.concat(latLngs[latLngs.length - 1])
-    else
-      latLngs = latLngs.concat(latLngs[0])
+    latLngs = latLngs.concat()
```
Comment (bilts) [inline app/assets/javascripts/modules/map/leaflet-plugins/interpolation.js.coffee]: Any idea what happened here? It looks like you're re-applying my polygon rendering fixes.

File: app/assets/javascripts/models/data/collection.js.coffee (hunk 1 above)
Comment (mightynimble) [inline app/assets/javascripts/models/data/collection.js.coffee]: Without the "?", canFocus will not always be eval to a boolean value. Therefore the granule filter button will not show up for cwic collection.

File: app/views/search/_granule_search.html.erb [hunk 4]
```diff
@@ -1,97 +1,116 @@
...
-        or
-        <a href="#" class="search-granules-multiple button-small button-outline-light">
-          Search Multiple <i class="fa fa-arrow-circle-o-right"></i>
-        </a>
-      </section>
-      <section class="multiple-granule-id-search">
-        <p>
-          <a href="#" class="search-granules-one button-small button-outline-light">
-            <i class="fa fa-arrow-circle-o-left"></i> Search One
+    <!--ko ifnot: cwic()-->
```
Comment (bilts) [inline app/views/search/_granule_search.html.erb]: I'd really like to find a better way to do this, long-term. The whole panel could/should probably be swapped.

File: config/services.yml [hunk 5]
```diff
@@ -4,19 +4,19 @@ earthdata:
     urs_root: "https://urs.earthdata.nasa.gov"
     echo_root: "https://api.echo.nasa.gov"
     cmr_root: "https://cmr.earthdata.nasa.gov" # Does not exist yet
-    opensearch_root: "https://api.echo.nasa.gov/opensearch"
+    opensearch_root: "https://cmr.earthdata.nasa.gov/opensearch"
```
Comment (bilts) [inline config/services.yml]: Again, shouldn't this already be in master?

File: app/assets/javascripts/modules/banner.coffee (hunk 2 above)
Comment (mightynimble) [inline app/assets/javascripts/modules/banner.coffee]: I have seen an error in the console twice. So I just added it there but I didn't investigate why the error occurs...

File: spec/features/cwic/granules/cwic_granule_filtering_spec.rb [hunk 6]
```diff
@@ -0,0 +1,50 @@
+require "spec_helper"
+
+describe "CWIC Granule list", reset: false do
```
Comment (bilts) [inline spec/features/cwic/granules/cwic_granule_filtering_spec.rb]: All acceptance criteria should have a matching spec that is tagged "acceptance: true"

File: app/assets/javascripts/modules/map/leaflet-plugins/interpolation.js.coffee (hunk 3 above)
Comment (mightynimble) [inline app/assets/javascripts/modules/map/leaflet-plugins/interpolation.js.coffee]: I rebased the branch against master, which may pull in some commits.

File: spec/features/data_access/service_access_spec.rb [hunk 7]
```diff
@@ -66,23 +66,21 @@
...
-        # this test and the one below are quite flaky.
-        # Cannot reliably display a progress bar using recordings
-        xit 'displays a progress bar while the service is processing' do
-          wait_for_xhr
-          if page.has_content?("AMSR-E/Aqua 5-Day L3 Global Snow Water Equivalent EASE-Grids V002 Processing")
-            expect(page).to have_content('of 12 items processed')
-            expect(page).to have_css('div.progress-bar')
-          else
-            expect(page).to have_content("AMSR-E/Aqua 5-Day L3 Global Snow Water Equivalent EASE-Grids V002 Complete")
+        end
```
Comment (bilts) [inline spec/features/data_access/service_access_spec.rb]: Another case of something that I thought was in master.

File: app/views/search/_granule_search.html.erb (hunk 4 above)
Comment (mightynimble) [inline app/views/search/_granule_search.html.erb]: Totally!!

File: spec/features/cwic/granules/cwic_granule_filtering_spec.rb (hunk 6 above)
Comment (mightynimble) [inline spec/features/cwic/granules/cwic_granule_filtering_spec.rb]: I will add the tag. Thanks!

File: app/assets/javascripts/modules/banner.coffee (hunk 2 above)
Comment (bilts) [inline app/assets/javascripts/modules/banner.coffee]: Please investigate or file a bug. This just ends up hiding a legitimate underlying problem.

**Pull request merged**